### ISSUE TITLE
Check dims prior to handing to Stan

### ISSUE THREAD

**Issue by mespe** [stan-dev/rstanarm#79]
I ran into issues when scaling predictors/responses. If the scale function built into R is used, the result is of dims [N,1] and Stan chokes. Using a custom function that returns a vector works fine.

It would be nice to either:
1) Include a scale function in rstanarm that plays nice with Stan, or
2) Check the data prior to handing off to Stan for dim mis-matches.

2 seems preferable.

**Comment by jgabry** [comment]
Thanks for reporting this. I find R's `scale` function to be rather annoying. The default method returns a matrix even if the input is a vector. I'd much prefer that it return an object of the same type as the input, but oh well.  

Anyway, I agree that your suggestion (2) is preferable to (1).

**Comment by jgabry** [comment]
@mespe To follow up, which of the rstanarm modeling functions were you using when you got these errors?

**Comment by bgoodri** [comment]
I agree that we should do (2) but would emphasize that using `scale` on the outside is not recommended. Internally, the functions in rstanarm center the predictors automatically. But dividing by the estimated standard deviation really just makes it harder to get posterior predictors right. It is sensible to divide variables by constants to get reasonable units, but even this should not be necessary for `stan_lm`, `stan_aov`, and `stan_polr`.

**Comment by jgabry** [comment]
@bgoodri I'm pretty sure only `stan_[g]lmer` has this issue. And I think the fix is just to change 

```{r} 
y <- glmod$fr[, as.character(glmod$formula[2L])]
```
to this

```{r} 
y <- model.response(glmod$fr)
```

The latter will drop dimensions but the former doesn't

**Comment by bgoodri** [comment]
OK

**Comment by jgabry** [comment]
@mespe I think this is fixed by 69fcb48, but if you still see this problem after updating we can definitely reopen this

**Comment by jgabry** [comment]
Better fix for this in bfa2b09
